app/src/main/java/com/example/hackernews/factories/RepositoryFactory.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.hackernews.factories
+
+import com.example.hackernews.common.constants.Api
+import com.example.hackernews.data.service.NewsService
+import com.example.hackernews.model.repository.NewsRepository
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object RepositoryFactory {
```

Comment:
Factory, as the name suggests, is the place where objects are created. They are beneficial for a couple of reasons:
- the logic of object creation is centralized
- objects can easily be made as singletons and shared  
- moves the responsibility of creating object away from the main application